hipchat/notifications.py

```diff
@@ -109,11 +111,14 @@ def _api(
     assert color in VALID_COLORS, "Invalid color value: {}".format(color)
     assert message_format in VALID_FORMATS, "Invalid format: {}".format(message_format)
 
-    token = _token()
     if token is None:
-        logger.debug("HipChat API token not found, logging message instead:")
-        logger.debug(message)
-        return
+        # try to read the toekn from the environment variables
+        token = _token()
+        if token is None:
+            logger.debug("HipChat API token not found, logging message instead:")
+            logger.debug(message)
+            return
+            
```

Comment:
Instead of having the two nested `if..` statements, you could combine them in to one by changing `token = _token()` into `token = token or _token()`:

You could combine these two `if..` statements into one using:

```python
    token = token or _token()
    if token is None:
        logger.debug("HipChat API token not found, logging message instead:")
        logger.debug(message)
        return
```

Reply:
Yes sure, definitely more elegant